addons/account_peppol/models/res_company.py

```diff
@@ -0,0 +1,239 @@
+            if not company._check_peppol_endpoint_number(PEPPOL_ENDPOINT_RULES):
+                raise ValidationError(_("The Peppol endpoint identification number is not correct."))
+
+    @api.constrains('peppol_purchase_journal_id')
+    def _check_peppol_purchase_journal_id(self):
+        for company in self:
+            if company.peppol_purchase_journal_id and company.peppol_purchase_journal_id.type != 'purchase':
+                raise ValidationError(_("A purchase journal must be used to receive Peppol documents."))
+
+    # -------------------------------------------------------------------------
+    # COMPUTE METHODS
+    # -------------------------------------------------------------------------
+
+    @api.depends('account_peppol_proxy_state')
+    def _compute_peppol_purchase_journal_id(self):
+        for company in self:
+            if not company.peppol_purchase_journal_id and company.account_peppol_proxy_state not in ('not_registered', 'rejected'):
+                company.peppol_purchase_journal_id = self.env['account.journal'].search([
+                    ('company_id', '=', company.id),
+                    ('type', '=', 'purchase'),
+                ], limit=1)
+                company.peppol_purchase_journal_id.is_peppol_journal = True
+            else:
+                company.peppol_purchase_journal_id = company.peppol_purchase_journal_id
+
+    def _inverse_peppol_purchase_journal_id(self):
+        for company in self:
+            # This avoid having 2 or more journals from the same company with
+            # `is_peppol_journal` set to True (which could occur after changes).
+            journals_to_reset = self.env['account.journal'].search([
+                ('company_id', '=', company.id),
+                ('is_peppol_journal', '=', True),
+            ])
+            journals_to_reset.is_peppol_journal = False
+            company.peppol_purchase_journal_id.is_peppol_journal = True
+
+    @api.depends('email')
+    def _compute_account_peppol_contact_email(self):
+        for company in self:
+            if not company.account_peppol_contact_email:
+                company.account_peppol_contact_email = company.email
+
+    @api.depends('phone')
+    def _compute_account_peppol_phone_number(self):
+        for company in self:
+            if not company.account_peppol_phone_number:
+                try:
+                    # precompute only if it's a valid phone number
+                    company._sanitize_peppol_phone_number(company.phone)
+                    company.account_peppol_phone_number = company.phone
+                except ValidationError:
+                    continue
+
+    # -------------------------------------------------------------------------
+    # LOW-LEVEL METHODS
+    # -------------------------------------------------------------------------
+
+    @api.model
+    def _sanitize_peppol_endpoint(self, vals, eas=False, endpoint=False):
+        # TODO: remove in master
```

Comment:
This method is not used anymore, only there for stable policy.
You can remove it.